Source/relay/TourGuide/Sets/Fax.ash

```diff
@@ -2,6 +2,33 @@
 string [int] SFaxGeneratePotentialFaxes(boolean suggest_less_powerful_faxes, boolean [monster] blocked_monsters)
 {
     string [int] potential_faxes;
+
+    record potential_fax {
```

Comment:
generally i think this is a solid structure. you are right that this is much, much heavier weight than the old form but i think this kind of record structure is largely fine. one thought though is that i think records need to be instantiated outside of your function here. it might work like this but i am pretty sure we almost always instantiate records outside of the scope of the function they're used in for a reason, though i am not remembering the exact reason offhand right now...

Reply:
FWIW, we do this in at least a couple places. For example: https://github.com/loathers/TourGuide/blob/4966227a4c8a4d1541804d02106c7dfc19e43821/Source/relay/TourGuide/Items%20of%20the%20Month/2020/Superhero%20Cape.ash#L23-L27

My best guess is that you can't return those record values from the function, but we don't return them in this case. I'm a big fan of making it function local as it avoids some of the horrible support guts leaking out.

I'll delay twiddling with this until I've got the other bits updated and poke around in Discord. Happy to move it if you've still got reservations.

Reply:
good spot! generally i prefer to instantiate them outside but i did not make that tile and the fact that the retrocape tile works makes me think my worry is just my preference not an actual need. fine to keep it within, thanks for doing the dive! :+1: